src/main/java/org/apache/flink/state/benchmark/StateBenchmarkBase.java

```diff
@@ -52,10 +54,25 @@
     final ThreadLocalRandom random = ThreadLocalRandom.current();
 
     @Param({"HEAP", "ROCKSDB"})
-    protected StateBackendBenchmarkUtils.StateBackendType backendType;
+    private StateBackendBenchmarkUtils.StateBackendType backendType;
+
+    @Param({StateBenchmarkConstants.defaultStateDataDirPath})
+    private String stateDataDirPath;
```

Comment:
Won't this in combination with https://github.com/apache/flink-benchmarks/blob/master/save_jmh_result.py#L72:L76 change the names of the uploaded benchmarks to the [code speed](http://codespeed.dak8s.net:8000/changes/)?

Do we really want this to be reported like this? Won't this clog the results and the WebUI in particular?
Secondly, even if we want this, we would have to manually rename all of the benchmarks in the database to preserve continuity of the results in the graphs.

Maybe it's actually better idea after all, to not use jmh's `@Param` for this? Maybe simply use an environmental variable?

Reply:
I didn't notice it. Now I prefer to not use `@Param` for this . Instead of using environmental variable, Can we add a benchmark conf as I mentioned in [FLINK-24967](https://issues.apache.org/jira/projects/FLINK/issues/FLINK-24967)

Reply:
I think either of those options would work, depends on what we want to achieve? I think using `@Params` makes sense if we really want to present the results from different configurations side by side, especially in the WebUI or in the continuous regression testing scripts. However if there are many many different constants that you would like to change, it probably doesn't make sense to keep them all separately as `@Params`, as it would simply explode the number of tests (jmh is testing all combinations of params). 

Having said all of that, a dedicated config file might be a not a bad solution. For now there can be a single configuration file that anyone can overwrite as he chooses. If we ever decide to present the results from different configs side by side, we can do this via `@Param` `preset`, where `preset` would be a name of one of those config files.

Reply:
ok, I will provide a single configuration implementation .